eng/pipelines/templates/steps/build-artifacts.yml

```diff
@@ -58,7 +58,17 @@ steps:
     parameters:
       ServiceDirectory: ${{ parameters.ServiceDirectory }}
       BuildTargetingString: ${{ parameters.BuildTargetingString }}
-
+  - task: Powershell@2
+    inputs:
+      filePath: $(Build.SourcesDirectory)/eng/common/scripts/Save-Package-Properties.ps1
```

Comment:
This is going to overwrite the package info files we just wrote in set-dev-build.yml. Should we instead have a parameter that adds the DevVersion all the time or do we need them written twice?

Reply:
If we only do it once, then Version and DevVersion are always the same thing.
Like -addDevVersion in the first step "Dump package property"
It was like
```
{
  "Name": "azure-template",
  "Version": "0.1.0b1108289",
  "DevVersion": "0.1.0b1108289",
  "DirectoryPath": "sdk/template/azure-template",
  "ServiceDirectory": "template",
  "ReadMePath": "sdk/template/azure-template/README.md",
  "ChangeLogPath": "sdk/template/azure-template/CHANGELOG.md",
  "Group": null,
  "SdkType": "client",
  "IsNewSdk": true,
  "ArtifactName": "azure-template",
  "ReleaseStatus": "2021-09-22"
}
```
If only have the Save-Package-Property run after 'update the dev version step' step
It was like
```
{
  "Name": "azure-template",
  "Version": "0.1.0a20210922003",
  "DevVersion": "0.1.0a20210922003",
  "DirectoryPath": "sdk/template/azure-template",
  "ServiceDirectory": "template",
  "ReadMePath": "sdk/template/azure-template/README.md",
  "ChangeLogPath": "sdk/template/azure-template/CHANGELOG.md",
  "Group": null,
  "SdkType": "client",
  "IsNewSdk": true,
  "ArtifactName": "azure-template",
  "ReleaseStatus": null
}

It is not ideal for python case. Also JS is running Save-Package-Property twice as well.
```

Reply:
As for other properties, they keep the same thing, overwrite does not affect the accuracy. Each json file is super tiny and one build only takes several json files, so write it twice is not a big burden to pipeline builds.

Reply:
OK I think I understand what is happening now. Just for confirmation can you show me what the json file looks like after running it twice?